src/index.spec.ts

```diff
@@ -2,50 +2,313 @@ import { renderHook } from '@testing-library/react-hooks';
+            useFetchWithRedux(testAction, testSelector, {
+              timeTillCacheInvalidate: 1234,
+            }),
+          );
+
+          expect(result.current).toEqual(null);
+          expect(mockDispatch).toHaveBeenCalledTimes(1);
+          expect(mockDispatch).toHaveBeenCalledWith(testAction());
+        });
+
+        it('Returns a value when useSelector returns a value, and should not dispatch an action', () => {
+          mockUseSelector.mockImplementation(callback =>
+            callback(testSelector(testSelector<number>(7))),
+          );
+
+          const { result } = renderHook(() =>
+            useFetchWithRedux(testAction, testSelector),
+          );
+
+          expect(result.current).toEqual(7);
+          expect(mockDispatch).not.toHaveBeenCalled();
+        });
+
+        it('Sets cache with the action type, time and cacheInvalidationTime from the hook input', () => {
+          mockUseSelector.mockImplementation(callback =>
+            callback(testSelector(testSelector<number>(7))),
+          );
+
+          renderHook(() =>
+            useFetchWithRedux(testAction, testSelector, {
+              timeTillCacheInvalidate: 1234,
+            }),
+          );
+
+          expect(setCacheTimeouts).toHaveBeenCalledTimes(1);
+          expect(setCacheTimeouts).toHaveBeenCalledWith({
+            TEST: {
+              cacheInvalidationTime: 1234,
+              cacheSet: Date.now(),
+            },
+          });
+        });
+      });
+    });
   });
 
-  it('Returns value from state and does not call dispatch when selector returns data', () => {
-    mockUseSelector.mockImplementation(callback =>
-      callback(testSelector<number>(7)),
-    );
+  describe('When cache has been set', () => {
+    beforeEach(() => {
+      mockUseContext.mockImplementation(() => ({
+        cacheTimeouts: {
+          TEST: {
+            cacheInvalidationTime: 1234,
+            cacheSet: Date.now(),
+          },
+        },
+        setCacheTimeouts,
+        timeTillCacheInvalidateGlobal: 2222,
+      }));
+
+      mockUseSelector.mockImplementation(callback =>
+        callback(testSelector(4444)),
+      );
+    });
+
+    describe('Before it has expired', () => {
+      beforeEach(() => {
+        mockHasCacheTimedOut.mockImplementation(() => false);
+      });
+
+      it('Returns the value from useSelector', () => {
+        const { result } = renderHook(() =>
+          useFetchWithRedux(testAction, testSelector),
+        );
+
+        expect(result.current).toEqual(4444);
+      });
+
+      it('Does not dispatch an action', () => {
+        renderHook(() => useFetchWithRedux(testAction, testSelector));
+        expect(mockDispatch).toHaveBeenCalledTimes(0);
+      });
+
+      it('Should not make a call to setCacheTimeouts to update the cache', () => {
+        renderHook(() => useFetchWithRedux(testAction, testSelector));
+        expect(setCacheTimeouts).toHaveBeenCalledTimes(0);
```

Comment:
I think: `expect(setCacheTimeouts).not.toHaveBeenCalled();` reads a little bit better when dealing with something that we are asserting to not have been called at all

Reply:
There will probably be other instances of that in this PR